subprojects/core/src/main/java/org/gradle/internal/service/scopes/CoreBuildSessionServices.java

```diff
@@ -155,26 +150,6 @@ BuildStartedTime createBuildStartedTime(Clock clock, BuildRequestMetaData buildR
         return BuildStartedTime.startingAt(Math.min(currentTime, buildRequestMetaData.getStartTime()));
     }
 
-    @Provides
-    protected ExecFactory decorateExecFactory(
```

Comment:
❓  Why must we move it down to the `BuildTree` scope?

If we indeed need a build-tree-scoped version, it's okay. But removing it from here has a side-effect that all consumers of the service in the `BuildSession` scope will now start getting an `UserHome` instance as their dependency. Maybe there are no such consumers, but we should be aware of that side effect.

If there are such consumers, we might want to keep the `BuildSession`-scope instance.

Reply:
Yeah, that is a problem. We either move ExecFactory to `BuildTreeScope` or we move other services to `BuildSessionScope` (e.g. ObjectFactory&friends). 

I think in that case "scope private services by default would help a lot".

Reply:
Can you explain why we need to move all services to the same scope and why it has to be a move, and not an additional service at the build-tree scope?

Reply:
I actually removed `ExecOperations` in Global scope and made it scope private in GradleUserHome. I think that solves it. It seems we don't need it in build session. 

But let's wait a bit what will tests say

Reply:
Ok, it seems ExecOperations is needed in BuildSessionScope too